http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -366,12 +366,17 @@ class GKEInterpreter[F[_]](
 
       // Resolve the cluster in Google
       googleClusterOpt <- gkeService.getCluster(gkeClusterId)
-      googleCluster <- F.fromOption(
-        googleClusterOpt,
-        ClusterCreationException(ctx.traceId,
-                                 s"Cluster not found in Google: ${gkeClusterId} | trace id: ${ctx.traceId}"
-        )
-      )
+      googleCluster <- googleClusterOpt match {
+        case Some(value) => F.pure(value)
+        case None =>
+          kubernetesClusterQuery.markAsDeleted(dbCluster.id, ctx.now).transaction >>
```

Comment:
this fixes an issue where if a GKE cluster is deleted in google, all Leo does is prompting user that the cluster doesn't exist.

With this change, we're marking the cluster as `DELETED` when leo encounters the issue, and user can try to create a new app again, and leo will just create new app.

This change makes it 100 times easier to work with GKE apps for development, and also benefits real users.